cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -242,10 +243,9 @@ public InetAddress providesHostname(CConfiguration cConf) {
     );
   }
 
-  private ProgramId getProgramIdFromRequest(ApplicationId preview, AppRequest request) {
-    // TODO remove this when UI passes the program name and type in the AppRequest.
+  private ProgramId getProgramIdFromRequest(ApplicationId preview, AppRequest request) throws BadRequestException {
     if (request.getPreview() == null) {
-      return preview.workflow("DataPipelineWorkflow");
+      throw new BadRequestException("preview config cannot be null");
```

Comment:
If someone starts the preview from UI, this can never be null, but if someone starts a preview using REST, this can be null, so I think it needs to be checked